hotel/models/hotel.py

```diff
@@ -399,11 +399,10 @@ def onchange_dates(self):
         @param self: object pointer
         @return: Duration and checkout_date
         '''
-        company_obj = self.env['res.company']
         configured_addition_hours = 0
-        company_ids = company_obj.search([])
-        if company_ids.ids:
-            configured_addition_hours = company_ids[0].additional_hours
+        whouse_com_id = self.warehouse_id.company_id
+        if self.warehouse_id and whouse_com_id:
+            configured_addition_hours = whouse_com_id.additional_hours
```

Comment:
This is a wrong code. 
you are using self.warehouse_id.company_id first and then checking the self.warehouse_id condition.